<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Givanna Putri" />

<meta name="date" content="2023-05-24" />

<title>Mitigating Batch Effects in the Integration of Multi-Batch Cytometry Data at the Supercell Level</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SuperCellCyto_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/phipsonlab/SuperCellCyto-analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Mitigating Batch Effects in the Integration
of Multi-Batch Cytometry Data at the Supercell Level</h1>
<h4 class="author">Givanna Putri</h4>
<h4 class="date">2023-05-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>SuperCellCyto-analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed42code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(42)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed42code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(42)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomphipsonlabSuperCellCytoanalysistree7baf7b743b8e6601fb3a3ddd87ccbd24bd3fb965targetblank7baf7b7a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/phipsonlab/SuperCellCyto-analysis/tree/7baf7b743b8e6601fb3a3ddd87ccbd24bd3fb965" target="_blank">7baf7b7</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomphipsonlabSuperCellCytoanalysistree7baf7b743b8e6601fb3a3ddd87ccbd24bd3fb965targetblank7baf7b7a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/phipsonlab/SuperCellCyto-analysis/tree/7baf7b743b8e6601fb3a3ddd87ccbd24bd3fb965" target="_blank">7baf7b7</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/b_cell_identification/.DS_Store
    Ignored:    code/batch_correction/.DS_Store
    Ignored:    code/explore_supercell_purity_clustering/.DS_Store
    Ignored:    code/explore_supercell_purity_clustering/functions/.DS_Store
    Ignored:    code/explore_supercell_purity_clustering/louvain_all_cells/.DS_Store
    Ignored:    code/label_transfer/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    data/bodenmiller_cytof/
    Ignored:    data/explore_supercell_purity_clustering/
    Ignored:    data/haas_bm/
    Ignored:    data/oetjen_bm_dataset/
    Ignored:    data/trussart_cytofruv/
    Ignored:    output/.DS_Store
    Ignored:    output/bodenmiller_cytof/
    Ignored:    output/explore_supercell_purity_clustering/
    Ignored:    output/label_transfer/
    Ignored:    output/oetjen_b_cell_panel/
    Ignored:    output/trussart_cytofruv/

Untracked files:
    Untracked:  Factor plot - Gated_Population - plotted on UMAP_X by UMAP_Y.png
    Untracked:  code/b_cell_identification/cluster_singlecell.R
    Untracked:  code/b_cell_identification/rescale_supercell.R
    Untracked:  code/b_cell_identification/runtime_benchmark/
    Untracked:  code/batch_correction/benchmark_cycombine.R
    Untracked:  code/batch_correction/rescale_supercell.R
    Untracked:  code/batch_correction/runtime_benchmark/
    Untracked:  code/bodenmiller_data/benchmark_supercell_runtime.R
    Untracked:  code/compare_da_test_runtime.R
    Untracked:  code/krieg_melanoma/
    Untracked:  code/label_transfer/harmony_knn_singlecell.R
    Untracked:  code/label_transfer/seurat_rpca_singlecell.R
    Untracked:  data.tar.gz
    Untracked:  output/krieg_melanoma/
    Untracked:  run_time_v1.Rmd

Unstaged changes:
    Modified:   SuperCellCyto-analysis.Rproj
    Modified:   code/batch_correction/run_batch_correction_supercells.R
    Modified:   code/label_transfer/harmony_knn.R
    Modified:   code/label_transfer/seurat_rpca.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/batch_correction.Rmd</code>) and
HTML (<code>docs/batch_correction.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/phipsonlab/SuperCellCyto-analysis/a55c3ba46180436d36d909f1ceb49772c673858c/docs/batch_correction.html" target="_blank">a55c3ba</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/phipsonlab/SuperCellCyto-analysis/366514efdb396f62d0f049c7b67885add6915856/docs/batch_correction.html" target="_blank">366514e</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/phipsonlab/SuperCellCyto-analysis/blob/402358ba3fe541410137e367c29468c07e7659a4/analysis/batch_correction.Rmd" target="_blank">402358b</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-07-28
</td>
<td>
wflow_publish(c("analysis/*Rmd"))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In the analysis, we explore the potential for reducing batch effects
at the supercell level. We employed 2 different batch correction
methodologies for this purpose: CytofRUV <span
class="citation">(Trussart et al. 2020)</span> and cyCombine <span
class="citation">(Pedersen et al. 2022)</span>.</p>
<p>The dataset used in this analysis originates from the CytofRUV study.
It consists of Peripheral Blood Mononuclear Cell (PBMC) samples from 3
Chronic Lymphocytic Leukaemia (CLL) patients and 9 Healthy Controls
(HC). Each patient’s sample was duplicated across two batches, with each
batch comprising 12 samples, totaling 24 samples. All samples were
stained with a 31-antibody panel, targeting 19 lineage markers and 12
functional proteins, and subsequently quantified using Cytof. The
complete list of the antibodies used can be found in the CytofRUV
manuscript <span class="citation">(Trussart et al. 2020)</span>.</p>
<p>For this analysis, we employed the following steps. First, we
downloaded the raw FCS files from FlowRepository <a
href="http://flowrepository.org/id/FR-FCM-Z2L2">FR-FCM-Z2L2</a>.
Following this, the data pre-processed using the R script provided with
the CytofRUV manuscript (the script is accessible in
<code>code/batch_correction/prepare_data.R</code>). Once the
pre-processing was completed, we generated supercells using
SuperCellCyto (R script is available in
<code>code/batch_correction/run_supercell.R</code>). Subsequently, we
applied CytofRUV and cyCombine to the supercells. Scripts facilitating
these corrections are available in
<code>code/batch_correction/run_batch_correction_supercells.R</code>.</p>
<p>The findings reported here represent the results following the
application of SuperCellCyto and batch correction methods.</p>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>library(here)
library(data.table)
library(limma)
library(ggplot2)
library(ggrepel)
library(Spectre)
library(pheatmap)
library(scales)
library(ggpubr)
library(RColorBrewer)
library(ggridges)</code></pre>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<p>Sample metadata.</p>
<pre class="r"><code>sample_metadata &lt;- fread(here(&quot;data&quot;, &quot;trussart_cytofruv&quot;, &quot;metadata&quot;, &quot;Metadata.csv&quot;))</code></pre>
<p>Markers.</p>
<pre class="r"><code>panel_info &lt;- fread(here(&quot;data&quot;, &quot;trussart_cytofruv&quot;, &quot;metadata&quot;, &quot;Panel.csv&quot;))
panel_info[, reporter_marker := paste(fcs_colname, antigen, &quot;asinh&quot;, &quot;cf5&quot;, sep = &quot;_&quot;)]

markers &lt;- panel_info$reporter_marker

cell_type_markers &lt;- panel_info[panel_info$marker_class == &quot;type&quot;]$reporter_marker</code></pre>
<p>All cells.</p>
<pre class="r"><code>cell_dat &lt;- fread(here(&quot;output&quot;, &quot;trussart_cytofruv&quot;, &quot;20230515_supercell_out&quot;, &quot;cell_dat_asinh.csv&quot;))
# Remove the untransformed markers as these are useless
cell_dat &lt;- cell_dat[, c(32:68)]
cell_dat[, batch := factor(paste0(&quot;batch&quot;, batch))]</code></pre>
<p>Uncorrected supercells.</p>
<pre class="r"><code>supercell_exp_mat &lt;- fread(here(&quot;output&quot;, &quot;trussart_cytofruv&quot;, &quot;20230524&quot;, &quot;supercellExpMat_rep1_umap.csv&quot;))
supercell_exp_mat[, batch := factor(paste0(&quot;batch&quot;, batch))]</code></pre>
<p>Corrected supercells.</p>
<pre class="r"><code>batch_corrected_supercells &lt;- list()
batch_corrected_supercells[[&quot;CytofRUV&quot;]] &lt;- fread(here(&quot;output&quot;, &quot;trussart_cytofruv&quot;, &quot;20230524&quot;, &quot;supercellExpMat_postCytofRUV.csv&quot;))
batch_corrected_supercells[[&quot;CytofRUV&quot;]][, batch := factor(paste0(&quot;batch&quot;, batch))]

batch_corrected_supercells[[&quot;cyCombine&quot;]] &lt;- fread(here(&quot;output&quot;, &quot;trussart_cytofruv&quot;, &quot;20230704&quot;, &quot;supercellExpMat_postCycombine.csv&quot;))
batch_corrected_supercells[[&quot;cyCombine&quot;]][, batch := paste0(&quot;batch&quot;, batch)]
setnames(batch_corrected_supercells$cyCombine, &quot;sample&quot;, &quot;sample_id&quot;)</code></pre>
</div>
<div id="assess-batch-correction-result" class="section level1">
<h1>Assess batch correction result</h1>
<div id="mds-plot" class="section level2">
<h2>MDS plot</h2>
<p>We evaluate the performance of the batch correction approaches by
generating MDS plots.</p>
<pre class="r"><code>mds_plots &lt;- list()

mds_plots[[1]] &lt;- make.mds.plot(
  dat = cell_dat,
  sample_col = &quot;sample_id&quot;,
  markers = markers,
  colour_by = &quot;batch&quot;,
  font_size = 2
) + labs(title = &quot;Uncorrected cells&quot;, colour = &quot;Batch&quot;) +
  scale_color_manual(values = c(&quot;batch1&quot; = &quot;#0047AB&quot;, &quot;batch2&quot; = &quot;#DC143C&quot;))

mds_plots[[2]] &lt;- make.mds.plot(
  dat = supercell_exp_mat,
  sample_col = &quot;sample_id&quot;,
  markers = markers,
  colour_by = &quot;batch&quot;,
  font_size = 2
) + labs(title = &quot;Uncorrected supercells&quot;, colour = &quot;Batch&quot;) +
  scale_color_manual(values = c(&quot;batch1&quot; = &quot;#0047AB&quot;, &quot;batch2&quot; = &quot;#DC143C&quot;))

mds_plots[[3]] &lt;- make.mds.plot(
  dat = batch_corrected_supercells$CytofRUV,
  sample_col = &quot;sample_id&quot;,
  markers = markers,
  colour_by = &quot;batch&quot;,
  font_size = 2
) + labs(title = &quot;CytofRUV corrected supercells&quot;, colour = &quot;Batch&quot;) +
  scale_color_manual(values = c(&quot;batch1&quot; = &quot;#0047AB&quot;, &quot;batch2&quot; = &quot;#DC143C&quot;))

mds_plots[[4]] &lt;- make.mds.plot(
  dat = batch_corrected_supercells$cyCombine,
  sample_col = &quot;sample_id&quot;,
  markers = markers,
  colour_by = &quot;batch&quot;,
  font_size = 2
) + labs(title = &quot;cyCombine corrected supercells&quot;, colour = &quot;Batch&quot;) +
  scale_color_manual(values = c(&quot;batch1&quot; = &quot;#0047AB&quot;, &quot;batch2&quot; = &quot;#DC143C&quot;))</code></pre>
<pre class="r"><code>ggarrange(plotlist = mds_plots, ncol = 2, nrow = 2, common.legend = TRUE, legend = &quot;bottom&quot;)</code></pre>
<p><img src="figure/batch_correction.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/phipsonlab/SuperCellCyto-analysis/blob/366514efdb396f62d0f049c7b67885add6915856/docs/figure/batch_correction.Rmd/unnamed-chunk-8-1.png" target="_blank">366514e</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In the MDS plots of uncorrected supercells, the 1st dimension
separates the CLL samples from the HC samples, while the 2nd dimension
separates the two batches. After applying batch correction methods, the
separation of CLL and HC samples are maintained along the 1st dimension.
However, the 2nd dimension, post-correction, no longer separate the
batches.</p>
</div>
<div id="umap-plot" class="section level2">
<h2>UMAP plot</h2>
<p>Complementing the MDS plots, we will also the UMAP plots of the
supercells coloured by the batches.</p>
<pre class="r"><code>umap_plt_dat &lt;- rbind(
  batch_corrected_supercells$CytofRUV[, c(&quot;UMAP_X&quot;, &quot;UMAP_Y&quot;, &quot;batch&quot;, &quot;patient_id&quot;)],
  batch_corrected_supercells$cyCombine[, c(&quot;UMAP_X&quot;, &quot;UMAP_Y&quot;, &quot;batch&quot;, &quot;patient_id&quot;)],
  supercell_exp_mat[, c(&quot;UMAP_X&quot;, &quot;UMAP_Y&quot;, &quot;batch&quot;, &quot;patient_id&quot;)]
)
umap_plt_dat$correction &lt;- c(
  rep(&quot;CytofRUV&quot;, nrow(batch_corrected_supercells$CytofRUV)),
  rep(&quot;cyCombine&quot;, nrow(batch_corrected_supercells$cyCombine)),
  rep(&quot;None&quot;, nrow(supercell_exp_mat))
)

ggplot(umap_plt_dat, aes(x = UMAP_X, y = UMAP_Y, colour = batch)) +
  geom_point(size = 0.3, alpha = 0.3) +
  facet_grid(cols = vars(patient_id), rows = vars(correction)) +
  scale_color_manual(values = c(&quot;batch1&quot; = &quot;#F8766D&quot;, &quot;batch2&quot; = &quot;#00BFC4&quot;)) +
  theme_classic() +
  guides(colour = guide_legend(override.aes = list(size = 5, alpha = 1))) +
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/batch_correction.Rmd/unnamed-chunk-9-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/phipsonlab/SuperCellCyto-analysis/blob/366514efdb396f62d0f049c7b67885add6915856/docs/figure/batch_correction.Rmd/unnamed-chunk-9-1.png" target="_blank">366514e</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The UMAP plots demonstrate that following batch correction,
supercells from different batches overlap more so compared to those that
are not corrected, further indicating the effective reduction of batch
effect.</p>
</div>
<div id="computing-earth-mover-distance" class="section level2">
<h2>Computing Earth Mover Distance</h2>
<p>The Earth Mover’s Distance (EMD) metric is often employed to measure
the efficacy of batch effect correction algorithm. It can be applied to
each marker to measure the dissimilarity in their distributions across
multiple batches. Following successful batch correction, the EMD score
for any marker is expected to decrease.</p>
<p>To assess the effectiveness of CytofRUV and cyCombine, for each
paired sample (indicated by the patient_id), we obtain distribution of
each marker, by binning the data into bins of size 0.1, from either all
cells or supercells. Thereafter, for each marker we calculate EMD
comparing the distribution between the paired sample.</p>
<pre class="r"><code>calc_emd &lt;- function(dat, markers, patient_ids) {
  emd_dist &lt;- lapply(patient_ids, function(pid) {
    emd_per_marker &lt;- lapply(markers, function(marker) {
      exp &lt;- dat[patient_id == pid, c(&quot;batch&quot;, marker), with = FALSE]
      hist_bins &lt;- lapply(c(&quot;batch1&quot;, &quot;batch2&quot;), function(bat_id) {
        exp_bat &lt;- exp[batch == bat_id, ][[marker]]
        as.matrix(graphics::hist(exp_bat, breaks = seq(-100, 100, by = 0.1), plot = FALSE)$counts)
      })
      emdist::emd2d(hist_bins[[1]], hist_bins[[2]])
    })
    emd_per_marker &lt;- data.table(
      marker = markers,
      emd = unlist(emd_per_marker),
      patient_id = pid
    )
    return(emd_per_marker)
  })
  emd_dist &lt;- rbindlist(emd_dist)
  return(emd_dist)
}</code></pre>
<p>Calculate EMDs.</p>
<pre class="r"><code>patient_ids &lt;- unique(supercell_exp_mat$patient_id)

emd_uncorrected &lt;- calc_emd(supercell_exp_mat, markers, patient_ids)
emd_cytofruv &lt;- calc_emd(batch_corrected_supercells$CytofRUV, markers, patient_ids)
emd_cycombine &lt;- calc_emd(batch_corrected_supercells$cyCombine, markers, patient_ids)

emd_dist &lt;- rbindlist(list(emd_uncorrected, emd_cytofruv, emd_cycombine))
emd_dist[, correction := c(
  rep(&quot;Uncorrected supercells&quot;, nrow(emd_uncorrected)),
  rep(&quot;CytofRUV&quot;, nrow(emd_cytofruv)),
  rep(&quot;cyCombine&quot;, nrow(emd_cycombine))
)]
emd_dist[, emd_rounded := round(emd, digits = 4)]
emd_dist[, sqrt_emd := sqrt(emd_rounded)]</code></pre>
<pre class="r"><code>ggplot(emd_dist, aes(x = patient_id, y = sqrt_emd, colour = correction)) +
  geom_boxplot(outlier.size = 0.7) +
  theme_classic() +
  scale_color_manual(values = c(
    &quot;CytofRUV&quot; = &quot;#FFAA33&quot;,
    &quot;cyCombine&quot; = &quot;#0047AB&quot;,
    &quot;Uncorrected supercells&quot; = &quot;#A9A9A9&quot;
  )) +
  labs(
    y = &quot;Square Root of EMD&quot;, x = &quot;Patient ID&quot;, colour = &quot;Batch Correction Method&quot;,
    title = &quot;Comparison of EMD Across Corrected and Uncorrected Supercells&quot;,
    subtitle = &quot;EMD: Earth Mover Distance&quot;
  ) +
  scale_y_continuous(breaks = pretty_breaks(n = 10))</code></pre>
<p><img src="figure/batch_correction.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/phipsonlab/SuperCellCyto-analysis/blob/366514efdb396f62d0f049c7b67885add6915856/docs/figure/batch_correction.Rmd/unnamed-chunk-12-1.png" target="_blank">366514e</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In general, for all methods, we see a reduction in EMD score.</p>
</div>
</div>
<div id="compare-the-distribution-of-the-markers"
class="section level1">
<h1>Compare the distribution of the markers</h1>
<p>Just between the uncorrected and corrected supercells.</p>
<pre class="r"><code>all_data &lt;- list(
  CytofRUV = batch_corrected_supercells$CytofRUV,
  cyCombine = batch_corrected_supercells$cyCombine,
  uncorrected = supercell_exp_mat
)

all_data &lt;- rbindlist(all_data, idcol = &quot;correction&quot;, fill = TRUE)
all_data &lt;- all_data[, c(markers, &quot;batch&quot;, &quot;correction&quot;), with = FALSE]

all_data &lt;- melt(all_data, id.vars = c(&quot;batch&quot;, &quot;correction&quot;))

all_data[, variable := gsub(&quot;^[^_]*_&quot;, &quot;&quot;, gsub(&quot;_asinh_cf5&quot;, &quot;&quot;, variable))]</code></pre>
<p>Plot the distribution for CD14, CD45RA, and PUMA for the main
figure.</p>
<pre class="r"><code>subset_marker_dist &lt;- all_data[variable %in% c(&quot;CD14&quot;, &quot;CD45RA&quot;, &quot;PUMA&quot;), ]
# for ordering the panels
subset_marker_dist[, correction := factor(correction, levels = c(&quot;uncorrected&quot;, &quot;CytofRUV&quot;, &quot;cyCombine&quot;))]

ggplot(subset_marker_dist, aes(x = value, y = batch)) +
  geom_density_ridges(aes(fill = correction, color = correction), alpha = 0.3) +
  facet_grid(rows = vars(variable), cols = vars(correction)) +
  theme_ridges() +
  theme(strip.text.x = element_text(margin = margin(b = 3))) +
  scale_fill_manual(values = c(&quot;uncorrected&quot; = &quot;#F8766D&quot;, &quot;CytofRUV&quot; = &quot;#00BFC4&quot;, &quot;cyCombine&quot; = &quot;#7CAE00&quot;)) +
  scale_color_manual(values = c(&quot;uncorrected&quot; = &quot;#F8766D&quot;, &quot;CytofRUV&quot; = &quot;#00BFC4&quot;, &quot;cyCombine&quot; = &quot;#7CAE00&quot;)) +
  labs(x = &quot;Marker Expression&quot;, y = &quot;Batch&quot;, title = &quot;Distribution of Marker Expression&quot;)</code></pre>
<p><img src="figure/batch_correction.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/phipsonlab/SuperCellCyto-analysis/blob/366514efdb396f62d0f049c7b67885add6915856/docs/figure/batch_correction.Rmd/unnamed-chunk-14-1.png" target="_blank">366514e</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Plot the rest of the distribution out.</p>
<pre class="r"><code>marker_dist &lt;- all_data[!variable %in% c(&quot;CD14&quot;, &quot;CD45RA&quot;, &quot;PUMA&quot;), ]
marker_dist[, correction := factor(correction, levels = c(&quot;uncorrected&quot;, &quot;CytofRUV&quot;, &quot;cyCombine&quot;))]

ggplot(marker_dist, aes(x = value, y = batch)) +
  geom_density_ridges(aes(fill = correction, color = correction), alpha = 0.3) +
  facet_grid(rows = vars(variable), cols = vars(correction)) +
  theme_ridges() +
  theme(strip.text.x = element_text(margin = margin(b = 3)), strip.text.y = element_text(margin = margin(b = 3))) +
  scale_fill_manual(values = c(&quot;uncorrected&quot; = &quot;#F8766D&quot;, &quot;CytofRUV&quot; = &quot;#00BFC4&quot;, &quot;cyCombine&quot; = &quot;#7CAE00&quot;)) +
  scale_color_manual(values = c(&quot;uncorrected&quot; = &quot;#F8766D&quot;, &quot;CytofRUV&quot; = &quot;#00BFC4&quot;, &quot;cyCombine&quot; = &quot;#7CAE00&quot;)) +
  labs(x = &quot;Marker Expression&quot;, y = &quot;Batch&quot;, title = &quot;Distribution of Marker Expression&quot;)</code></pre>
<p><img src="figure/batch_correction.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/phipsonlab/SuperCellCyto-analysis/blob/366514efdb396f62d0f049c7b67885add6915856/docs/figure/batch_correction.Rmd/unnamed-chunk-15-1.png" target="_blank">366514e</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-pedersen2022cycombine" class="csl-entry">
Pedersen, Christina Bligaard, Søren Helweg Dam, Mike Bogetofte Barnkob,
Michael D Leipold, Noelia Purroy, Laura Z Rassenti, Thomas J Kipps, et
al. 2022. <span>“cyCombine Allows for Robust Integration of Single-Cell
Cytometry Datasets Within and Across Technologies.”</span> <em>Nature
Communications</em> 13 (1): 1698.
</div>
<div id="ref-trussart2020removing" class="csl-entry">
Trussart, Marie, Charis E Teh, Tania Tan, Lawrence Leong, Daniel HD
Gray, and Terence P Speed. 2020. <span>“Removing Unwanted Variation with
CytofRUV to Integrate Multiple CyTOF Datasets.”</span> <em>Elife</em> 9:
e59630.
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Monterey 12.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggridges_0.5.4     RColorBrewer_1.1-3 ggpubr_0.6.0       scales_1.2.1      
 [5] pheatmap_1.0.12    Spectre_1.0.0-0    ggrepel_0.9.3      ggplot2_3.4.1     
 [9] limma_3.54.1       data.table_1.14.8  here_1.0.1         workflowr_1.7.0   

loaded via a namespace (and not attached):
  [1] backports_1.4.1             plyr_1.8.8                 
  [3] igraph_1.4.0                lazyeval_0.2.2             
  [5] sp_1.6-0                    splines_4.2.3              
  [7] flowCore_2.10.0             listenv_0.9.0              
  [9] scattermore_0.8             GenomeInfoDb_1.34.9        
 [11] digest_0.6.31               htmltools_0.5.4            
 [13] viridis_0.6.2               fansi_1.0.4                
 [15] magrittr_2.0.3              tensor_1.5                 
 [17] cluster_2.1.4               ROCR_1.0-11                
 [19] globals_0.16.2              matrixStats_0.63.0         
 [21] cytolib_2.10.0              spatstat.sparse_3.0-0      
 [23] colorspace_2.1-0            xfun_0.39                  
 [25] dplyr_1.1.0                 callr_3.7.3                
 [27] RCurl_1.98-1.10             jsonlite_1.8.4             
 [29] progressr_0.13.0            spatstat.data_3.0-0        
 [31] survival_3.5-3              zoo_1.8-11                 
 [33] glue_1.6.2                  polyclip_1.10-4            
 [35] gtable_0.3.1                zlibbioc_1.44.0            
 [37] XVector_0.38.0              leiden_0.4.3               
 [39] DelayedArray_0.24.0         car_3.1-1                  
 [41] future.apply_1.10.0         SingleCellExperiment_1.20.0
 [43] BiocGenerics_0.44.0         abind_1.4-5                
 [45] DBI_1.1.3                   rstatix_0.7.2              
 [47] spatstat.random_3.1-3       miniUI_0.1.1.1             
 [49] Rcpp_1.0.10                 viridisLite_0.4.1          
 [51] xtable_1.8-4                reticulate_1.28            
 [53] rsvd_1.0.5                  stats4_4.2.3               
 [55] htmlwidgets_1.6.1           httr_1.4.4                 
 [57] ellipsis_0.3.2              Seurat_4.3.0               
 [59] ica_1.0-3                   farver_2.1.1               
 [61] pkgconfig_2.0.3             sass_0.4.5                 
 [63] uwot_0.1.14                 deldir_1.0-6               
 [65] utf8_1.2.3                  labeling_0.4.2             
 [67] tidyselect_1.2.0            rlang_1.0.6                
 [69] reshape2_1.4.4              later_1.3.0                
 [71] munsell_0.5.0               tools_4.2.3                
 [73] cachem_1.0.6                cli_3.6.0                  
 [75] generics_0.1.3              broom_1.0.3                
 [77] evaluate_0.20               stringr_1.5.0              
 [79] fastmap_1.1.0               yaml_2.3.7                 
 [81] goftest_1.2-3               processx_3.8.0             
 [83] knitr_1.42                  fs_1.6.1                   
 [85] fitdistrplus_1.1-8          purrr_1.0.1                
 [87] RANN_2.6.1                  pbapply_1.7-0              
 [89] future_1.31.0               nlme_3.1-162               
 [91] whisker_0.4.1               mime_0.12                  
 [93] compiler_4.2.3              rstudioapi_0.14            
 [95] plotly_4.10.1               png_0.1-8                  
 [97] ggsignif_0.6.4              emdist_0.3-2               
 [99] spatstat.utils_3.0-1        tibble_3.1.8               
[101] bslib_0.4.2                 stringi_1.7.12             
[103] highr_0.10                  ps_1.7.2                   
[105] lattice_0.20-45             Matrix_1.5-3               
[107] vctrs_0.5.2                 pillar_1.8.1               
[109] lifecycle_1.0.3             spatstat.geom_3.0-6        
[111] lmtest_0.9-40               jquerylib_0.1.4            
[113] RcppAnnoy_0.0.20            cowplot_1.1.1              
[115] bitops_1.0-7                irlba_2.3.5.1              
[117] httpuv_1.6.9                patchwork_1.1.2            
[119] GenomicRanges_1.50.2        R6_2.5.1                   
[121] promises_1.2.0.1            RProtoBufLib_2.10.0        
[123] KernSmooth_2.23-20          gridExtra_2.3              
[125] IRanges_2.32.0              parallelly_1.34.0          
[127] codetools_0.2-19            MASS_7.3-58.2              
[129] SummarizedExperiment_1.28.0 rprojroot_2.0.3            
[131] withr_2.5.0                 SeuratObject_4.1.3         
[133] sctransform_0.3.5           S4Vectors_0.36.1           
[135] GenomeInfoDbData_1.2.9      parallel_4.2.3             
[137] grid_4.2.3                  tidyr_1.3.0                
[139] rmarkdown_2.20              carData_3.0-5              
[141] MatrixGenerics_1.10.0       Rtsne_0.16                 
[143] git2r_0.31.0                getPass_0.2-2              
[145] spatstat.explore_3.0-6      Biobase_2.58.0             
[147] shiny_1.7.4                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
