<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Givanna Putri" />

<meta name="date" content="2023-05-24" />

<title>Recovery of Differentially Expressed Cell State Markers Across Stimulated and Unstimulated Human Peripheral Blood Cells</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SuperCellCyto_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Recovery of Differentially Expressed Cell
State Markers Across Stimulated and Unstimulated Human Peripheral Blood
Cells</h1>
<h4 class="author">Givanna Putri</h4>
<h4 class="date">2023-05-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-07-28
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>SuperCellCyto-analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed42code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(42)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed42code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(42)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong402358b">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 402358b
</a>
</p>
</div>
<div id="strongRepositoryversionstrong402358b"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 402358b.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/bodenmiller_cytof/
    Ignored:    data/explore_supercell_purity_clustering/
    Ignored:    data/haas_bm/
    Ignored:    data/oetjen_bm_dataset/
    Ignored:    data/trussart_cytofruv/
    Ignored:    output/.DS_Store
    Ignored:    output/bodenmiller_cytof/
    Ignored:    output/explore_supercell_purity_clustering/
    Ignored:    output/label_transfer/
    Ignored:    output/oetjen_b_cell_panel/
    Ignored:    output/trussart_cytofruv/

Untracked files:
    Untracked:  README.html
    Untracked:  analysis/ref.bib
    Untracked:  code/README.html
    Untracked:  code/b_cell_identification/
    Untracked:  code/batch_correction/
    Untracked:  code/bodenmiller_data/
    Untracked:  code/explore_supercell_purity_clustering/
    Untracked:  code/label_transfer/
    Untracked:  data/README.html
    Untracked:  output/README.html

Unstaged changes:
    Modified:   .gitignore
    Modified:   README.md
    Modified:   _workflowr.yml
    Modified:   analysis/_site.yml
    Modified:   code/README.md
    Modified:   data/README.md
    Modified:   output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/de_test.Rmd</code>) and HTML
(<code>docs/de_test.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
402358b
</td>
<td>
Givanna Putri
</td>
<td>
2023-07-28
</td>
<td>
wflow_publish(c("analysis/*Rmd"))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this analysis, we assess whether a differential expression
analysis performed at the supercell level can recapitulate previously
published findings obtained by performing differential expression
analysis using the Diffcyt algorithm <span class="citation">(Weber et
al. 2019)</span> at the single cell level.</p>
<p>Specifically, we analysed a publicly available mass cytometry dataset
quantifying the immune cells in stimulated and unstimulated human
peripheral blood cells (BCR_XL dataset <span
class="citation">(Bodenmiller et al. 2012)</span>). This is a paired
experimental design, with each of the 8 independent samples, obtained
from 8 different individuals, contributing to both stimulated and
unstimulated samples (16 samples in total).</p>
<p>This dataset was previously analysed using Diffcyt to identify the
cell state markers that were differentially expressed between the
stimulated samples (BCR-XL group) and the unstimulated samples
(Reference group).</p>
<p>Our aim was to replicate these findings using a combination of
SuperCellCyto and the Limma R package <span class="citation">(Ritchie et
al. 2015)</span>. The analysis protocol is as the following. First the
data was downloaded using the HDCytoData package and an arcsinh
transformation with cofactor 5 was applied. Subsequently, SuperCellCyto
was applied on the data. The relevant scripts can be found in
<code>code/bodenmiller_data/download_bodenmiller_data.R</code> and
<code>code/bodenmiller_data/run_supercell_bodenmiller_data.R</code>.</p>
<p>The findings reported here represent the results following the
application of SuperCellCyto.</p>
</div>
<div id="load-libraries-and-data" class="section level1">
<h1>Load libraries and data</h1>
<pre class="r"><code>library(data.table)
library(ggplot2)
library(limma)
library(pheatmap)
library(gridExtra)
library(here)
library(SuperCell)

supercell_cell_map &lt;- fread(here(&quot;output&quot;, &quot;bodenmiller_cytof&quot;, &quot;20230522&quot;, &quot;supercell_gamma20_cell_map.csv&quot;))
cell_info &lt;- fread(here(&quot;output&quot;, &quot;bodenmiller_cytof&quot;, &quot;20230522&quot;, &quot;cell_info_with_cell_id.csv&quot;))
supercell_mat &lt;- fread(here(&quot;output&quot;, &quot;bodenmiller_cytof&quot;, &quot;20230522&quot;, &quot;supercell_gamma20_exp_mat.csv&quot;))
markers &lt;- fread(here(&quot;data&quot;, &quot;bodenmiller_cytof&quot;, &quot;markers_info.csv&quot;))

supercell_cell_map &lt;- merge.data.table(
  x = supercell_cell_map,
  y = cell_info[, c(&quot;cell_id&quot;, &quot;population_id&quot;), with = FALSE],
  by.x = &quot;CellId&quot;,
  by.y = &quot;cell_id&quot;
)</code></pre>
</div>
<div id="examine-purity-of-supercells" class="section level1">
<h1>Examine purity of supercells</h1>
<p>Purity was calculated using the implementation provided by the
SuperCell package.</p>
<pre class="r"><code>supercell_purity &lt;- as.data.table(
  supercell_purity(clusters = supercell_cell_map$population_id,
  supercell_membership = supercell_cell_map$SuperCellID),
  keep.rownames = TRUE
)
setnames(supercell_purity, c(&quot;V1&quot;, &quot;V2&quot;), c(&quot;supercell_id&quot;, &quot;purity_score&quot;))

ggplot(supercell_purity, aes(x = purity_score)) +
  geom_histogram(binwidth = .05) +
  labs(
    x = &quot;Purity score&quot;,
    y = &quot;Number of supercells&quot;,
    title = &quot;Supercell purity for gamma = 20&quot;,
    subtitle = paste(&quot;Mean purity =&quot;, round(mean(supercell_purity$purity_score), 2))
  ) +
  theme_bw() +
  scale_y_continuous(breaks = scales::pretty_breaks(n=10)) +
  scale_x_continuous(breaks = scales::pretty_breaks(n=10))</code></pre>
<p><img src="figure/de_test.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-expression-analysis" class="section level1">
<h1>Differential Expression Analysis</h1>
<p>We first annotate the supercells based on the most abundant cell type
they contain. In situations where the most abundant cell type is a tie
between several different cell types, the one that appears first in the
sequence is used as the supercell’s annotation.</p>
<pre class="r"><code>supercell_annotation &lt;- supercell_cell_map[, .N, by = .(SuperCellID, population_id)][
  order(-N), .SD[1], by = SuperCellID
]

supercell_mat &lt;- merge.data.table(
  x = supercell_mat,
  y = supercell_annotation[, c(&quot;SuperCellID&quot;, &quot;population_id&quot;)],
  by.x = &quot;SuperCellId&quot;,
  by.y = &quot;SuperCellID&quot;
)</code></pre>
<p>Subsequently, we compute the mean expression of each state marker
across different sample and cell type.</p>
<pre class="r"><code># Do DE test on just the cell state markers
markers_state &lt;- paste0(markers[markers$marker_class == &#39;state&#39;]$marker_name, &quot;_asinh_cf5&quot;)

supercell_mean_perSampPop &lt;- supercell_mat[, lapply(.SD, mean), by = c(&quot;sample_id&quot;, &quot;population_id&quot;), .SDcols = markers_state]

# Add in the group details and the patient id
sample_info &lt;- unique(cell_info, by = c(&quot;group_id&quot;, &quot;patient_id&quot;, &quot;sample_id&quot;))
sample_info[, c(&quot;cell_id&quot;, &quot;population_id&quot;) := list(NULL, NULL)]

supercell_mean_perSampPop &lt;- merge.data.table(supercell_mean_perSampPop, sample_info, by = &quot;sample_id&quot;)
supercell_mean_perSampPop[, `:=`(
  group_id = factor(group_id, levels=c(&quot;Reference&quot;, &quot;BCR-XL&quot;)),
  patient_id = factor(patient_id, levels=paste0(&quot;patient&quot;, seq(8))),
  sample_id = factor(sample_id)
)]
supercell_mean_perSampPop &lt;- supercell_mean_perSampPop[order(group_id)]</code></pre>
<p>For each cell type, we then used the limma package to identify state
markers that are differentially expressed across the 2 conditions.
Specifically, we take into account the paired experimental design and
use the <code>treat</code> function with <code>fc=1.1</code> to compute
empirical Bayes moderated-t p-values relative to the minimum fold-change
threshold set using the <code>fc</code> parameter.</p>
<pre class="r"><code>cell_types &lt;- c(&quot;B-cells IgM+&quot;, &quot;CD4 T-cells&quot;, &quot;NK cells&quot;)

limma_res_supercell &lt;- lapply(cell_types, function(ct) {

  pseudobulk_sub &lt;- supercell_mean_perSampPop[population_id == ct]

  groups &lt;- pseudobulk_sub$group_id
  
  # Taking into account the paired measurement
  patients &lt;- pseudobulk_sub$patient_id
  design &lt;- model.matrix(~0 + groups + patients)
  colnames(design) &lt;- gsub(&quot;groups&quot;, &quot;&quot;, colnames(design))
  colnames(design) &lt;- gsub(&quot;-&quot;, &quot;_&quot;, colnames(design))
  cont &lt;- makeContrasts(
    RefvsBCRXL = Reference - BCR_XL,
    levels = colnames(design)
  )
  fit &lt;- lmFit(t(pseudobulk_sub[, markers_state, with = FALSE]), design)
  fit_cont &lt;- contrasts.fit(fit, cont)
  fit_cont &lt;- eBayes(fit_cont, trend=TRUE, robust=TRUE)
  treat_all &lt;- treat(fit_cont, fc=1.1)
  
  res &lt;- as.data.table(
    topTreat(treat_all, number = length(markers_state)),
    keep.rownames = TRUE
  )
  # rename the rowname column as marker rather than &quot;rn&quot;
  setnames(res, &quot;rn&quot;, &quot;marker&quot;)
  res[, cell_type := ct]
  
  return(res)
})
limma_res_supercell &lt;- rbindlist(limma_res_supercell)</code></pre>
<p>Finally we use heatmap to assess the results returned by limma.</p>
<pre class="r"><code># 1 panel per population
pheatmap_sample_info &lt;- copy(sample_info)
pheatmap_sample_info[, `:=`(
  group_id = factor(group_id, levels = c(&quot;Reference&quot;, &quot;BCR-XL&quot;)),
  patient_id = factor(patient_id, levels = paste0(&quot;patient&quot;, seq_len(8)))
)]
setnames(pheatmap_sample_info, &quot;group_id&quot;, &quot;Group&quot;)
pheatmap_sample_info &lt;- pheatmap_sample_info[order(Group, patient_id)]
pheatmap_sample_info[, patient_id := NULL]

pheatmap_sample_info &lt;- data.frame(pheatmap_sample_info)
rownames(pheatmap_sample_info) &lt;- pheatmap_sample_info$sample_id
pheatmap_sample_info$sample_id &lt;- NULL

cell_types_to_plot &lt;- c(&quot;B-cells IgM+&quot;, &quot;CD4 T-cells&quot;, &quot;NK cells&quot;)

heatmaps &lt;- lapply(cell_types_to_plot, function(ct) {
  # state markers status based on limma result
  markers_states_info &lt;- limma_res_supercell[cell_type == ct,]
  markers_states_info[, FDR_sig := factor(ifelse(adj.P.Val &lt;= 0.05, &quot;Yes&quot;, &quot;No&quot;), levels = c(&quot;Yes&quot;, &quot;No&quot;)) ]
  markers_states_info[, marker := gsub(&quot;_asinh_cf5&quot;, &quot;&quot;, marker)]
  markers_states_info &lt;- markers_states_info[order(FDR_sig)]

  markers_states_info &lt;- data.frame(markers_states_info[, c(&quot;marker&quot;, &quot;FDR_sig&quot;)])
  rownames(markers_states_info) &lt;- markers_states_info$marker
  markers_states_info$marker &lt;- NULL

  # filter to keep only a population,
  # change the ordering of rows (samples) to suit sample_info so &quot;reference&quot;
  # group comes first,
  # set sample_id as row names
  # rename the columns to remove _asinh
  # reorganise the columns so the significant ones come first
  pheatmap_data &lt;- supercell_mean_perSampPop[population_id == ct,]
  pheatmap_data[, sample_id := factor(sample_id, levels = rownames(pheatmap_sample_info))]
  rownames(pheatmap_data) &lt;- pheatmap_data$sample_id
  pheatmap_data[, population_id := NULL]
  setnames(pheatmap_data, markers_state, gsub(&quot;_asinh_cf5&quot;, &quot;&quot;, markers_state))

  pheatmap_data[, c(&quot;patient_id&quot;, &quot;group_id&quot;) := list(NULL, NULL)]
  setcolorder(pheatmap_data, c(&quot;sample_id&quot;, rownames(markers_states_info)))
  
  pheatmap_mat &lt;- as.matrix(pheatmap_data, rownames = &quot;sample_id&quot;)

  pheatmap(
    mat = pheatmap_mat,
    cluster_cols = FALSE,
    annotation_row = pheatmap_sample_info,
    annotation_col = markers_states_info,
    main = ct,
    cluster_rows = FALSE,
    annotation_colors = list(
      Group = c(&quot;Reference&quot; = &quot;#E6E6FA&quot;, &quot;BCR-XL&quot; = &quot;#5D3FD3&quot;),
      FDR_sig = c(&quot;Yes&quot; = &quot;darkgreen&quot;, &quot;No&quot; = &quot;grey&quot;)
    ),
    scale = &quot;column&quot;
  )
})</code></pre>
<p><img src="figure/de_test.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/de_test.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/de_test.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Our findings were consistent with those identified by Diffcyt,
including elevated expression of pS6, pPlcg2, pErk, and pAkt in B cells
in the stimulated group, along with reduced expression of pNFkB in the
stimulated group.</p>
<p>We also recapitulated the Diffcyt results in CD4 T cells and Natural
Killer (NK) cells, with significant differences in the expression of
pBtk and pNFkB in CD4 T cells between the stimulated and unstimulated
groups, and distinct differences in the expression of pBtk, pSlp76, and
pNFkB in NK cells between the stimulated and unstimulated groups.</p>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-bodenmiller2012multiplexed" class="csl-entry">
Bodenmiller, Bernd, Eli R Zunder, Rachel Finck, Tiffany J Chen, Erica S
Savig, Robert V Bruggner, Erin F Simonds, et al. 2012.
<span>“Multiplexed Mass Cytometry Profiling of Cellular States Perturbed
by Small-Molecule Regulators.”</span> <em>Nature Biotechnology</em> 30
(9): 858–67.
</div>
<div id="ref-ritchie2015limma" class="csl-entry">
Ritchie, Matthew E, Belinda Phipson, DI Wu, Yifang Hu, Charity W Law,
Wei Shi, and Gordon K Smyth. 2015. <span>“Limma Powers Differential
Expression Analyses for RNA-Sequencing and Microarray Studies.”</span>
<em>Nucleic Acids Research</em> 43 (7): e47–47.
</div>
<div id="ref-weber2019diffcyt" class="csl-entry">
Weber, Lukas M, Malgorzata Nowicka, Charlotte Soneson, and Mark D
Robinson. 2019. <span>“Diffcyt: Differential Discovery in
High-Dimensional Cytometry via High-Resolution Clustering.”</span>
<em>Communications Biology</em> 2 (1): 183.
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Monterey 12.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] SuperCell_1.0     here_1.0.1        gridExtra_2.3     pheatmap_1.0.12  
[5] limma_3.54.1      ggplot2_3.4.1     data.table_1.14.8 workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] statmod_1.5.0      tidyselect_1.2.0   xfun_0.39          bslib_0.4.2       
 [5] splines_4.2.3      colorspace_2.1-0   vctrs_0.5.2        generics_0.1.3    
 [9] htmltools_0.5.4    yaml_2.3.7         utf8_1.2.3         rlang_1.0.6       
[13] jquerylib_0.1.4    later_1.3.0        pillar_1.8.1       glue_1.6.2        
[17] withr_2.5.0        RColorBrewer_1.1-3 lifecycle_1.0.3    stringr_1.5.0     
[21] munsell_0.5.0      gtable_0.3.1       evaluate_0.20      knitr_1.42        
[25] callr_3.7.3        fastmap_1.1.0      httpuv_1.6.9       ps_1.7.2          
[29] fansi_1.0.4        highr_0.10         Rcpp_1.0.10        promises_1.2.0.1  
[33] scales_1.2.1       cachem_1.0.6       jsonlite_1.8.4     farver_2.1.1      
[37] fs_1.6.1           digest_0.6.31      stringi_1.7.12     processx_3.8.0    
[41] dplyr_1.1.0        getPass_0.2-2      rprojroot_2.0.3    grid_4.2.3        
[45] cli_3.6.0          tools_4.2.3        magrittr_2.0.3     sass_0.4.5        
[49] tibble_3.1.8       whisker_0.4.1      pkgconfig_2.0.3    rmarkdown_2.20    
[53] httr_1.4.4         rstudioapi_0.14    R6_2.5.1           git2r_0.31.0      
[57] compiler_4.2.3    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
