---
title: "SuperCellCyto analysis"
author:
  - name: Givanna Putri
date: "`r Sys.Date()`"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---

# Abstract {-}

The rapid advancements in cytometry technologies have enabled the quantification of up to 50 proteins across millions of cells at a single-cell resolution. The analysis of cytometry data necessitates the use of computational tools for tasks such as data integration, clustering, and dimensionality reduction. While numerous computational methods exist in the cytometry and single-cell RNA sequencing (scRNAseq) fields, many are hindered by extensive run times when processing large cytometry data containing millions of cells. Existing solutions, such as random subsampling, often prove inadequate as they risk excluding small, rare cell subsets. To address this, we propose a practical strategy that builds on the SuperCell framework from the scRNAseq field. The supercell concept involves grouping single cells with highly similar transcriptomic profiles, and has been shown to be an effective unit of analysis for scRNAseq data. We show that for cytometry datasets, there is no loss of information by grouping cells into supercells. Further, we demonstrate the effectiveness of our approach by conducting a series of downstream analyses on six publicly available cytometry datasets at the supercell level, and successfully replicating previous findings performed at the single cell level. We present a computationally efficient solution for transferring cell type labels from single-cell multiomics data which combines RNA with protein measurements, to a cytometry dataset, allowing for more precise cell type annotations. Our SuperCellCyto R package and the associated analysis workflows are available on our GitHub repositories (https://github.com/phipsonlab/SuperCellCyto and https://github.com/phipsonlab/SuperCellCyto-analysis/).

# Authors {-}

Givanna H. Putri^1^, George Howitt^2^, Felix Marsh-Wakefield^3^, Thomas Ashhurst^4^, Belinda Phipson^1^

^1^ The Walter and Eliza Hall Institute of Medical Research, Parkville, VIC, Australia

^2^ Peter MacCallum Cancer Centre and The Sir Peter MacCallum Department of Oncology, The University of Melbourne, Parkville, VIC, Australia

^3^ Centenary Institute of Cancer Medicine and Cell Biology, Sydney, NSW, Australia

^4^ Sydney Cytometry Core Research Facility and School of Medical Sciences, The University of Sydney, Sydney, NSW, Australia 

# Analysis Overview {-}
1. [Supercells Preserve Biological Heterogeneity and Facilitate Efficient Cell Type Identification](explore_supercell_purity_clustering.html)
2. [Supercells Combined with Clustering Can Identify Rare B Cell Subsets](b_cells_identification.html)
3. [Mitigating Batch Effects in the Integration of Multi-Batch Cytometry Data at the Supercell Level](batch_correction.html)
4. [Recovery of Differentially Expressed Cell State Markers Across Stimulated and Unstimulated Human Peripheral Blood Cells](de_test.html)
5. [Identification of Differentially Abundant Rare Monocyte Subsets in Melanoma Patients](da_test.html)
6. [Efficient Cell Type Label Transfer Between CITEseq and Cytometry Data](label_transfer.html)
7. [Analysis of Run Times](run_time.html)

# Updates

31st Jan 2023: Extra analyses were done to address reviewers' comments.
Code used to perform the extra analyses are available on the `code` directory
of this website's [Github repository](https://github.com/phipsonlab/SuperCellCyto-analysis/).

Extra analyses done and corresponding codes:

1. For [Supercells Preserve Biological Heterogeneity and Facilitate Efficient Cell Type Identification](explore_supercell_purity_clustering.html).
Codes are stored in `code/explore_supercell_purity_clustering/additional_code`.
    * `count_cell_type_per_supercell.R`: counted how many cell types were captured per supercell and illustrated that as bar chart.
    * `draw_purity_table.R`: stratify the purity scores and compute how many supercells obtained purity of 1, 0.9-1, 0.5-0.9, <0.5.
    * `plot_supercell_size_distribution.R`: illustrate the number of cells captured per supercell.
    * `randomly_group_cells.R`: randomly group cells into groups which total number matches the number of supercells generated, 
    and calculate base purity score.
2. For [Supercells Combined with Clustering Can Identify Rare B Cell Subsets](b_cells_identification.html).
Codes are stored in `code/b_cell_identification/additional_code`.
    * `analyse_subsampled_data.R`: randomly subsampled over 400k cells, clustered them, and attempted to identify the B cell subsets.
3. For [Mitigating Batch Effects in the Integration of Multi-Batch Cytometry Data at the Supercell Level](batch_correction.html).
Codes are stored in `code/batch_correction/additional_code`.
    * `calculate_scib_metrics_for_submission.ipynb`: python notebook for computing biological signal preservation metrics in scib package.
    * `cluster_supercells.R`: cluster the supercells before and after batch correction. Needed for computing the biological signal preservation metrics.
    * `create_h5ad_files.ipynb`: convert the output of the clustered supercelsl to h5ad file. Needed for computing the biological signal preservation metrics.
    * `plot_metrics.R`: plot the heatmap to illustrate the scores for biological signal preservation metrics.
4. For [Efficient Cell Type Label Transfer Between CITEseq and Cytometry Data](label_transfer.html).
Codes are stored in `code/label_transfer/additional_code`.
    * `calculate_weighted_accuracy.ipynb`: python notebook to compute weighted accuracy.
    * `check_subsets.R`: plotting the UMAP of the CITEseq annotations (original and mapped).
    * `confusion_matrix.R`: for drawing confusion matrices.
    * `draw_umap.R`: draw UMAP for the outcome of the label transfer.
    * `plot_accuracies.R`: plot charts for the accuracy scores.
    



